### ISSUE TITLE
Support of <Show Difference> at IDEs

### ISSUE THREAD

**Issue by Nikita-Guchakov** [robstoll/atrium#1882]
Hi, currently IDEs won't show <Click to show difference> if Atrium assertion fails
```kotlin
   expect(actual).toEqual(expected)
```

<img width="1521" alt="image" src="https://github.com/user-attachments/assets/5ead2591-8322-4f26-adae-c9e5158c9189" />

That makes test failures with large comparing objects annoying check.

To fix that you just need to move to throwing AssertionFailedError  from the https://github.com/ota4j-team/opentest4j JUnit initiative  instead of AtriumError (or maybe just extend AtriumError from it, haven't check Atrium internals)
like many other assertion libraries already do (junit, assertj, assertk, strikt, kotest, kotlintest, etc).

<img width="1522" alt="image" src="https://github.com/user-attachments/assets/77ad736e-52ab-4d8a-a760-6ced0efab8e3" />

**Comment by robstoll** [comment]
Thanks for the report.
TL;DR Atrium v1.3.0 will include it

Your feedback would be valuable here:
https://github.com/robstoll/atrium-roadmap/issues/39#issuecomment-2540074550

I'll keep this one open so that others see it as well

**Comment by robstoll** [comment]
I released v1.3.0-alpha-1 which includes a click to see difference link
https://github.com/robstoll/atrium/releases/tag/v1.3.0-alpha-1

Please try it out and give feedback, especially if you should find bugs or if something seems odd in reporting. Thanks
